main/javascript-modifications/ses-guide.md

```diff
@@ -0,0 +1,454 @@
+  console for debug information only. The console is not obliged to write to the POSIX standard output.
+
+- `lockdown()` and `harden()` both freeze an object’s API surface (enumerable data properties). 
+  A hardened object’s properties cannot be changed, only read, so the only way to interact with a 
+  hardened object is through its methods. `harden()` is similar to `Object.freeze()` but more 
+  thorough. See the individual [`lockdown()`](#lockdown) and [`harden()`](#harden) sections
+  below. 
+
+- `[Compartment](https://github.com/Agoric/SES-shim/tree/SES-v0.8.0/packages/ses#compartment)` is 
+  a global. Code runs inside a `Compartment` and can create sub-compartments to host other 
+  code (with different globals or transforms). Note that these child compartments get `harden()` and `Compartment`.
+
+## Realms
+
+Agoric deploy scripts and smart contract code run in an *immutable
+realm* with *Compartments* providing just enough authority to create
+useful and secure contracts. But not enough authority to do anything
+unintended or harmful to the participants of the smart contract. 
+
+JavaScript code runs in the context of 
+a [*Realm*](https://www.ecma-international.org/ecma-262/10.0/index.html#sec-code-realms). A 
+realm is the set of *primordials* (objects and standard library functions
+like `Array.prototype.push`) and a global object. In a web browser, an iframe is a realm. 
+In Node.js, a Node process is a realm.
+
+For historical reasons, the ECMAScript specification requires primordials
+be mutable (`Array.prototype.push = yourFunction` is valid ECMAScript but not 
+recommended). By using the Agoric SES shim and calling `lockdown()`, you can turn the 
+current realm into an *immutable realm*; a realm within which the primordials 
+are deeply frozen. 
+
+SES also lets programs create *Compartments*. These are "mini-realms".
+A Compartment has its own dedicated global object and environment, but 
+it inherits the primordials from their parent realm. Components are described
+in detail in the next section. 
+
+## Compartments
+A *compartment* is an execution environment for evaluating a stranger’s code. It has
+its own `globalThis` global object and wholly independent system of 
+modules. Otherwise it shares the same batch of intrinsics such as `Array` with its surrounding 
+compartment. The concept of a compartment implies an initial compartment, 
+the initial execution environment of a realm. After lockdown, all compartments share the same 
+frozen realm. 
+
+Here we create a compartment with a `print()` function on `globalThis`.
+```js
+import 'ses';
+
+const c = new Compartment({
+  print: harden(console.log),
+});
+
+c.evaluate(`
+  print('Hello! Hello?');
+`);
+```
+This new compartment has a different global object than the start compartment **tyg todo: Not sure what "start compartment"
+means? The initial compartment? One from the Zoe start() method?** . The 
```

Comment:
We posit that all JavaScript executes in a realm and compartment. Every realm has distinct intrinsics, whereas every compartment shares intrinsics. The initial realm and compartment are not constructed with `new Realm()` or `new Compartment()` but that’s invisible to the code running: they could just as well be running within a constructed realm or compartment. We call the one compartment in a realm that was not expressly constructed the *start compartment*. The start compartment receives some ambient authorities from the host, often access to timers and IO that are denied to other compartments. Running lockdown does not erase these powerful objects, but puts the program running in the start compartment on a footing where it is possible to carefully delegate powers to child compartments.